Source/WTF/wtf/text/CodePointIterator.h

```diff
@@ -35,57 +35,57 @@ template<typename CharacterType>
 class CodePointIterator {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    ALWAYS_INLINE CodePointIterator() { }
-    ALWAYS_INLINE CodePointIterator(const CharacterType* begin, const CharacterType* end)
-        : m_begin(begin)
-        , m_end(end)
+    ALWAYS_INLINE CodePointIterator() = default;
+    ALWAYS_INLINE CodePointIterator(std::span<const CharacterType> data)
+        : m_data(data)
     {
     }
     
     ALWAYS_INLINE CodePointIterator(const CodePointIterator& begin, const CodePointIterator& end)
-        : CodePointIterator(begin.m_begin, end.m_begin)
+        : CodePointIterator({ begin.m_data.data(), end.m_data.data() })
```

Comment:
Can’t this just be `begin.m_data` without the braces?

Reply:
I don't think so, this is essentially merging the spans from `begin` and `end` CodePointIterators. We can't ignore `end`.

Reply:
Oops, didn’t notice that!